integrations/lambda/demo.ipynb

```diff
@@ -1423,21 +1423,712 @@
+    "                \"timestream:ListTables\",\n",
+    "                \"timestream:DescribeDatabase\",\n",
+    "            ],\n",
+    "            \"Resource\": f\"arn:aws:timestream:{REGION_NAME}:{account_id}:database/{DATABASE_NAME}\"\n",
+    "        }\n",
+    "    ]\n",
+    "}\n",
+    "\n",
+    "try:\n",
+    "    # Attach the inline policy\n",
+    "    iam_client.put_role_policy(\n",
+    "        RoleName=workspace_role_name,\n",
+    "        PolicyName='GrafanaWorkspaceAccessPolicy',\n",
+    "        PolicyDocument=json.dumps(inline_policy)\n",
+    "    )\n",
+    "    print(f\"Attached inline policy to role {workspace_role_name}\")\n",
+    "except Exception as err:\n",
+    "    print(\"Failed to attach policy to workspace role\")\n",
+    "    raise"
+   ]
+  },
   {
    "cell_type": "markdown",
    "id": "112cf6a6",
    "metadata": {},
    "source": [
-    "### Grafana Configuration Steps\n",
-    "1. Open Grafana.\n",
-    "2. Add a new data source with the name \"timestream_sample_app_database\".\n",
-    "3. Create a new dashboard using the provided `dashboard.json`."
+    "### Create Workspace or Use Existing Workspace"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3b1d7609",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "WORKSPACE_NAME = \"sample_app_workspace\"\n",
+    "MAX_WAIT_SECONDS = 900 # 15 minutes\n",
+    "WAIT_PERIOD_SECONDS = 15\n",
+    "\n",
+    "workspace_id = \"\"\n",
+    "grafana_endpoint_url = \"\"\n",
+    "try:\n",
+    "    list_workspaces_response = grafana_client.list_workspaces()\n",
+    "    for workspace in list_workspaces_response['workspaces']:\n",
+    "        if workspace['name'] == WORKSPACE_NAME:\n",
+    "            print(f\"Workspace '{WORKSPACE_NAME}' already exists with ID: {workspace['id']}\")\n",
+    "            workspace_id = workspace['id']\n",
+    "            current_wait_seconds = 0\n",
+    "            if workspace['status'] != 'ACTIVE':\n",
+    "                status = \"\"\n",
+    "                while current_wait_seconds < MAX_WAIT_SECONDS:\n",
+    "                    status = grafana_client.describe_workspace(workspaceId=workspace_id)['workspace']['status']\n",
+    "                    print(f\"Workspace status: {status}\")\n",
+    "                    if status == 'ACTIVE':\n",
+    "                        break\n",
+    "                    time.sleep(WAIT_PERIOD_SECONDS)\n",
+    "                    current_wait_seconds += WAIT_PERIOD_SECONDS\n",
+    "                if current_wait_seconds >= MAX_WAIT_SECONDS and status != 'ACTIVE':\n",
+    "                    raise Exception(\"Timed out while waiting for workspace to become active\")\n",
+    "\n",
+    "except ClientError as e:\n",
+    "    raise Exception(f\"Error checking for workspace: {e}\")\n",
+    "\n",
+    "if not workspace_id:\n",
+    "    create_workspace_response = grafana_client.create_workspace(\n",
```

Comment:
You're right, this can be included when the workspace is being created. I have removed this cell and added the configuration when creating the workspace. I have also added more exception handling when creating the workspace.